api/extension-guides/ai/ai-extensibility-overview.md

```diff
@@ -0,0 +1,146 @@
+
+- **Code completion**: Offers inline code suggestions as you type
+- **Agent mode**: Enables AI to autonomously plan and execute development tasks with specialized tools
+- **Chat**: Lets developers use natural language to ask questions or make edits in codebase through chat interfaces
+- **Smart actions**: Use AI-enhanced actions for common development tasks, integrated throughout the editor
+
+You can extend and customize each of these built-in capabilities to create tailored AI experiences that meet the specific needs of your users.
+
+## Why extend AI in VS Code?
+
+Adding AI capabilities to your extension brings several benefits to your users:
+
+- **Enhanced user experience**: Provide intelligent assistance tailored to your extension's domain
+- **Domain specialization**: Create AI features specific to a programming language, framework, or domain
+- **Improved developer productivity**: Enhance common developer tasks, like debugging, code reviewing or testing, with AI capabilities
+- **Extend chat capabilities**: Add specialized tools or assistants to the chat interface for more powerful interactions
+
+## Extend the chat experience
+
+### Language model tool
+
+Language model tools enable you to extend agent mode in VS Code with domain-specific capabilities. In agent mode, these tools are automatically invoked based on the user's chat prompt to perform specialized tasks or retrieve information from a data source or service. Users can also reference these tools explicitly in their chat prompt by #-mentioning the tool.
+
+To implement a language model tool, use the [Language Model Tools API](/api/extension-guides/ai/tools) within your VS Code extension. A language model tool can access all VS Code extension APIs and provide deep integration with the editor.
+
+**Key benefits**:
+
+- Domain-specific capabilities as part of an autonomous coding workflow
+- Automatic tool invocation in agent mode, depending on user intent
+- Deep VS Code integration through extension APIs
+- Easy distribution and deployment via the Visual Studio Marketplace
+
+**Key considerations**:
+
+- Remote deployment requires the extension to implement the client-server communication
+- Reuse across different tools requires modular design and implementation
+
+### MCP tool
+
+Model Context Protocol (MCP) tools provide a way to integrate external services with language models by using a standardized protocol. In agent mode, these tools are automatically invoked based on the user's chat prompt to perform specialized tasks or retrieve information from external data sources.
+
+MCP tools run outside of VS Code, either locally on the user's machine or as a remote service. Users can add MCP tools through JSON configuration or VS Code extension can configure them programmatically. You can implement MCP tools through various language SDKs and deployment options.
+
+As MCP tools run outside of VS Code, they do not have access to the VS Code extension APIs.
+
+**Key benefits**:
+
```

Comment:
I would maybe try to cut down the Key Benefits list. Since a lot of repetation with Language Model Tools. Also a key benefit is - MCP server can be use in under MCP clients.